xclim/indices/_threshold.py

```diff
@@ -273,7 +335,63 @@ def continuous_snow_cover_end(
         .end
     )
     out.attrs.update(units="", is_dayofyear=np.int32(1), calendar=get_calendar(snd))
-    return out
+    return out.where(~valid)
+
+
+@declare_units(snw="[mass]/[area]", thresh="[mass]/[area]")
+def snw_season_end(
+    snw: xarray.DataArray,
+    thresh: Quantified = "20 kg m-2",
+    window: int = 14,
+    freq: str = "AS-JUL",
+) -> xarray.DataArray:
+    r"""End date of continuous snow water cover.
+
+    First day after the start of the continuous snow water cover when snow water is below a threshold (default: 2 cm)
+    for at least `N` (default: 14) consecutive days.
+
+    Warnings
+    --------
+    The default `freq` is valid for the northern hemisphere.
+
+    Parameters
+    ----------
+    snw : xarray.DataArray
+        Surface snow amount.
+    thresh : str
+        Threshold snow amount.
+    window : int
+        Minimum number of days with snow water below threshold.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray, [dimensionless]
+        First day after the start of the continuous snow water cover when the snow water
+        goes below a threshold for a minimum duration.
+        If there is no such day, returns np.nan.
+
+    References
+    ----------
+    :cite:cts:`chaumont_elaboration_2017`
+    """
+    from xclim.core.missing import (  # pylint: disable=import-outside-toplevel
+        at_least_n_valid,
+    )
```

Comment:
Why not just import this at the top?

Reply:
e.g. in `snd_max_doy`

Reply:
Maybe the justification in this case is that it's used only for this function? I still find this peculiar, didn't see anything like this elsewhere in XClim. I'll just put things in the header. I should  remove `# pylint: disable=import-outside-toplevel` too

Reply:
Ah yes, I remember now you told me about this

Reply:
Oh! I just checked and IIRC, the loops were caused by `core.missing` importing `indices.generic.select_time`. However, that function has since moved to `core.calendar`. Thus, it might now be safe to put the import at top level! Has to be tests, but at a quick glance, it seems it could work.

Reply:
I ran the test `TestSndMaxDoy` , seems to work!